src/Forwarder.cpp

```diff
@@ -216,9 +216,12 @@ void Forwarder::report_status() {
   using nlohmann::json;
   auto Status = json::object();
   auto Streams = json::array();
-  for (auto const &Stream : streams.getStreams()) {
-    Streams.push_back(Stream->getStatusJson());
-  }
+  auto StreamVector = streams.getStreams();
+  std::transform(StreamVector.cbegin(), StreamVector.cend(),
+                 std::back_inserter(Streams),
+                 [](const std::shared_ptr<Stream> &CStream) {
+                   return CStream->getStatusJson();
+                 });
```

Comment:
Interesting, was this a suggested change from clang-tidy or just your preference? In this situation, when the whole container is being looped over, I find the `for` loop more readable than a `transform` or `for_each`. It is a matter of personal preference though, I'm not suggesting you change it back.

Reply:
Cppcheck recently got the ability to recognise standard loops and to suggest an algorithm to use in its place. I decided to try to apply those suggestions mostly for the fun of it and for the learning experience. With that said, I agree that the original for-loop is easier to read to me. On the other hand, when searching the web for opinions from C++ experts, many suggest that you actually should use stl algorithms when possible.

Reply:
Oh nice, I should update my local cppcheck.